docs/docs/persistent-volumes.md

```diff
@@ -39,69 +43,77 @@ Additionally, you need to set a `residency` in order to tell Marathon to setup a
+If an agent re-registers with the cluster and offers its resources, Marathon is eventually able to relaunch a task there. If a node does not re-register with the cluster, Marathon will wait forever to receive expected offers, as it's goal is to re-use the existing data. If the agent is not expected to come back, you can manually delete the relevant tasks by adding a `wipe=true` flag and Marathon will eventually launch a new task with a new volume on another agent.
 
 ### Disk consumption
 
 As of Mesos 0.28, destroying a persistent volume will not cleanup or destroy data. Mesos will delete metadata about the volume in question, but the data will remain on disk. To prevent disk consumption, you should manually remove data when you no longer need it.
 
-### <a name="non-unique-roles"></a>Non-unique Roles
+<a name="non-unique-roles"></a>
+### Non-unique Roles
 
-Both static and dynamic reservations in Mesos are bound to roles, not to frameworks or framework instances. Marathon will add labels to claim that resources have been reserved for a combination of `frameworkId` and `taskId`, as noted above. However, these labels do not protect from misuse by other frameworks or old Marathon instances (prior to 1.0). Every Mesos framework that registers for a given role will eventually receive offers containing resources that have been reserved for that role. If such a framework does not respect the presence of labels and the sematic as intended by the creator of these labels, Marathon is unable to reclaim them and cannot act properly. It is therefore recommended to never use the same role for different frameworks if one of them uses dynamic reservations. Marathon instances in HA mode do not need to have unique roles, however, because they use the same role by design.
+Both static and dynamic reservations in Mesos are bound to roles, not to frameworks or framework instances. Marathon will add labels to claim that resources have been reserved for a combination of `frameworkId` and `taskId`, as noted above. However, these labels do not protect from misuse by other frameworks or old Marathon instances (prior to 1.0). Every Mesos framework that registers for a given role will eventually receive offers containing resources that have been reserved for that role.
+
+However, if a framework does not respect the presence of labels and the semantics as intended, Marathon is unable to reclaim the resources reserved for the role. We recommend never using the same role for different frameworks if one of them uses dynamic reservations. Marathon instances in HA mode do not need to have unique roles, though, because they use the same role by design.
```

Comment:
Labels do not have an implicit semantic; what they mean is up to the subject who defines them. Is this clear from the text?